doc/version_history.rst

```diff
@@ -2,6 +2,20 @@
 Version History
 ===============
 
+v0.4.1
+------
+* In CSC:
+  * Refactor how telemetry/event loops start/stop.
+  * Refactor handle_summary_state.
+  * Use begin_<cmd> to report state transition ack in progress for all state commands.
+* In tcp_client, don't write eof when closing connection and log when connection is closed.
+* Improve how mock_server closes tasks.
+* Improve how model closes tasks.
+* Remove usage of deprecated package `asynctest` in `test_csc`.
+* Only log queue full exception when it first occurs, then log when condition no longer present.
+* Remove `check_queue_size` utility method.
+* Update telemetry loop to wait for new messages instead of pooling.
```

Comment:
As I wrote in the another comment, the blocking call is used, and the application might not be able to close itself when blocked or the tcp buffer in kernel might be full.